src/overmap.cpp

```diff
@@ -601,6 +599,189 @@ void reset_region_settings()
+            jio.read("primary", region.default_groundcover_str->primary_str);
+        }
+        if (jio.has_member("secondary")) {
+            jio.read("secondary", region.default_groundcover_str->secondary_str);
+        }
+        if (jio.has_member("ratio")) {
+            jio.read("ratio", region.default_groundcover.chance);
+        }
+    }
+
+    if (jo.has_member("num_forests")) {
+        jo.read("num_forests", region.num_forests);
+    }
+    if (jo.has_member("forest_size_min")) {
+        jo.read("forest_size_min", region.forest_size_min);
+    }
+    if (jo.has_member("forest_size_max")) {
+        jo.read("forest_size_max", region.forest_size_max);
+    }
+    if (jo.has_member("house_basement_chance")) {
+        jo.read("house_basement_chance", region.house_basement_chance);
+    }
+    if (jo.has_member("swamp_maxsize")) {
+        jo.read("swamp_maxsize", region.swamp_maxsize);
+    }
+    if (jo.has_member("swamp_river_influence")) {
+        jo.read("swamp_river_influence", region.swamp_river_influence);
+    }
+    if (jo.has_member("swamp_spread_chance")) {
+        jo.read("swamp_spread_chance", region.swamp_spread_chance);
+    }
+
+    if (jo.has_object("field_coverage")) {
+        JsonObject pjo = jo.get_object("field_coverage");
+
+        if (pjo.has_member("percent_coverage")) {
+            double tmpval = 0.0f;
+            pjo.read("percent_coverage", tmpval);
+            region.field_coverage.mpercent_coverage = (int)(tmpval * 10000.0);
+        }
+
+        if (pjo.has_member("default_ter")) {
+            pjo.read("default_ter", region.field_coverage.default_ter_str);
+        }
+
+        if (pjo.has_object("other")) {
+            double tmpval = 0.0f;
+            JsonObject opjo = pjo.get_object("other");
+            std::set<std::string> keys = opjo.get_member_names();
+            for( const auto &key : keys ) {
+                tmpval = 0.0f;
+                if( key != "//" ) {
+                    if( opjo.read( key, tmpval ) ) {
+                        region.field_coverage.percent_str[key] = tmpval;
+                    }
+                }
+            }
+        }
+
+        if (pjo.has_member("boost_chance")) {
+            double tmpval = 0.0f;
+            pjo.read("boost_chance", tmpval);
+            if(tmpval != 0.0f) {
+                region.field_coverage.boost_chance = (int)(tmpval * 10000.0);
+
+                if (pjo.has_member("boosted_percent_coverage")) {
+                    pjo.read("boosted_percent_coverage", tmpval);
+                    region.field_coverage.boosted_mpercent_coverage = (int)(tmpval * 10000.0);
+                }
+                if(region.field_coverage.boosted_mpercent_coverage == 0.0f) {
+                    pjo.throw_error("boost_chance > 0 requires boosted_percent_coverage");
+                }
+
+                if (pjo.has_member("boosted_other_percent")) {
+                    pjo.read("boosted_other_percent", tmpval);
```

Comment:
In this situation, you can use the return value `JsonObject::read`. It returns true if the member was found, otherwise false. So it can be shorten to:
```C++
if( pjo.read( "...", tmpval ) ) {
    // do something with tmpval
```